src/components/carousel/index.tsx

```diff
@@ -212,6 +232,14 @@ const Carousel: FC<Props> = (props) => {
           paginationCarouselRef={paginationCarouselRef}
         />
       ) : null}
+
+      {hasNumbersPagination ? (
+        <NumbersPagination
+          mainCarousel={mainCarousel}
+          currentSlideIndex={selectedIndex}
+          numbers={props.children.map((_, index) => index)}
```

Comment:
as far as I can see, you are not doing anything with the elements of the array. You just need an array with a specific length. How about we pass: `numberOfSlides={props.children.length}`?